packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/components/VMConcernsCellRenderer.tsx

```diff
@@ -23,10 +25,11 @@ const categoryWeights = {
 };
 
 export const VMConcernsCellRenderer: React.FC<VMCellProps> = ({ data }) => {
+  const concerns = hasConcerns(data?.vm) ? data.vm.concerns : [];
```

Comment:
if you need this `if` it means you need separate renderer, the renderer should not need `if` s to know what type it's rendering, that is why we separate them

Reply:
@yaacov 
works as designed. [VMCellProps](https://github.com/kubev2v/forklift-console-plugin/blob/8c17951019a08a33f55b7c820e694551dd99c6dc/packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/components/VMCellProps.ts#L11) interface should work with all VMs. Since each provider have different fields this requires casts or type guards.  The type guard option is considered safer so I've used it above.
The alternative that I see is to parametrize the VMCellProps and VmData types by actual VM type. This approach was discussed and rejected before.

Reply:
works is not neccarily works as design, it works for you, but it has a gurd that should not be there, so it's not "as design" please remove the offending `hasConcerns` this method should work without testing for type or concerns, if you need to test for type (or the availability of concerns) it means you need to have special renderer for each type.

Reply:
limited scope of this PR only to adding `object` prop